ios/brave-ios/Sources/AIChat/Components/Input/AIChatPromptInputView.swift

```diff
@@ -220,6 +300,28 @@ struct AIChatPromptInputView: View {
       isNoMicrophonePermissionPresented = true
     }
   }
+
+  func scaleDown(image: UIImage, to targetSize: CGSize) -> UIImage {
+    if image.size.width <= targetSize.width && image.size.height <= targetSize.height {
+      return image
+    }
+
+    // Maintain Aspect Ratio
+    let widthRatio = targetSize.width / image.size.width
+    let heightRatio = targetSize.height / image.size.height
+    let scaleFactor = min(widthRatio, heightRatio)
+    let newSize = CGSize(
+      width: image.size.width * scaleFactor,
+      height: image.size.height * scaleFactor
+    )
+
+    let renderer = UIGraphicsImageRenderer(size: newSize)
+    let scaledImage = renderer.image { _ in
+      image.draw(in: CGRect(origin: .zero, size: newSize))
+    }
+
+    return scaledImage
+  }
```

Comment:
No need to do this manually, use https://developer.apple.com/documentation/uikit/uiimage/preparingthumbnail(of:)